docs/Config_Reference.md

```diff
@@ -1131,7 +1131,40 @@ the nature of skew correction these lengths are set via gcode. See
 [skew_correction]
 ```
 
-## Customized homing
+### [z_thermal_adjust]
+
+Temperature-dependant toolhead Z position adjustment. Compensate for vertical
+toolhead movement caused by thermal expansion of the printer's frame in
+real-time using a temperature sensor (typically coupled to a vertical section
+of frame).
+
+See also: [extended g-code commands](G-Codes.md#z_thermal_adjust).
+
+```
+[z_thermal_adjust]
+#temp_coeff:
+#   The temperature coefficient of expansion, in mm/K. For example, a
+#   temp_coeff of 0.01 mm/K will move the Z axis downwards by 0.01 mm for every
+#   Kelvin/degree Celsius that the temperature sensor increases. Defaults to
+#   0.0, which applies no adjustment.
```

Comment:
Please use only Celsius in the description.  I find it very confusing to use different units for different settings, so the convention in Klipper is for distances to always be in mm, temperatures in Celsius, time in seconds, etc.  I understand Kelvin and Celsius are interchangeable in this context, but for consistency I think the documentation should only use Celsius.